### DISCUSSION TITLE
Pre-generate menu

### DISCUSSION THREAD

**Discussion by QuarkAgency** [VirtusLab/strapi-plugin-navigation#363] [Ideas]
Hey,

When the menu become big, it takes a lot of effort for strapi to always generate the tree and always to provide the same one.
I propose a button to generate the tree (and other type) directly in the admin. It would generate a JSON file like : strapi-endpoint/navigation/1/tree.json.

In my case, Strapi take 2 seconds to provide the tree to my frontend.

**Comment by cyp3rius** [comment]
What I can propose in that case is cache on Navigation endpoints. We are using standard caching plugins with rules applied to navigation as well. With good results in couple commercial projects :)
